_code/InSituTree/vignettes/InSituTree_basicVignette.Rmd

```diff
@@ -0,0 +1,329 @@
+genes, only this gene would be considered as having high differentiating power
+at the given granularity level. Note that when fewer cell types or subtypes are
+considered simultaneously, we expect to see more variability in the diffs and
+percs.
+
+10. `return_summary_annotation`: A logical value indicating whether to return
+the final summary annotation.
+This data frame shows the annotation that has been made for each cell at each
+level of the annotation tree. Recommendation is to keep this at its default
+value of TRUE.
+
+```{r strReturnSummary}
+return_summary_df <- TRUE
+```
+
+## Tree-style cell typing
+
+First, we can print a simple structured list to visualize the cell type
+hierarchy we've laid out with our cell type hierarchy list.
+
+```{r cellTypeStructure, fig.width=6, fig.height=4}
+printTree(celltype_hierarchy_list, prefix = "|-")
+```
+
+Once we've verified that this tree shows the hierarchy and cell type
+relationships that we'd expect, we can run supervised nested cell typing.
+
+```{r runInSituTree}
+res <- runInSituTree(
+  x = counts,
+  neg = negmean,
+  full_profiles = ioprofiles,
+  cth = celltype_hierarchy_list,
+  name_for_new_annotation = annotation_names,
+  cohort = NULL,
+  excluded_genes = genes_to_exclude,
+  quantile_absolute_expression_difference_param = quantile_abs_difference,
+  quantile_percent_expression_difference_param = quantile_perc_difference,
+  return_summary_annotation = return_summary_df
+)
+```
+
+In the messages that print out, we see that a handful of cells are removed from
+cell typing if they have zero counts in any of the gene subsets used for typing.
+
+## Review results
+
+The results from running `InSituTree` contain two main components:
+
+1) Nested list of results from each run of `InSituTypeML`
+
+2) Summary of annotations at each level of the tree
+
+The nested results list contains the typical outputs of `InSituType` for each
+round of cell typing, and one additional metric, `ctsPerCell`, which is the 
+counts of the selected genes for each cell at the given round of subclustering.
+This counts metric could be used to evaluate the confidence of a particular
+round of subclustering, since cell type assignments for low-count cells are
+likely to be less confident. 
+Each level of the results list contains:
+
+- `result`: The `InSituType` results plus counts per cell
+
+- `subclusterings`: A nested list of clustering results for the next level down
+of annotations.
+
+- `name`: The name of the celltype that was subclustered.
+
+As an example, we look at the assignment of
+structural cells to fibroblast or endothelial
+and use the flightpath tool from `InSituType` to visualize how well cells were
+assigned to these major types.
+
+```{r resList}
+names(res)
+
+names(res$topLevel)
+
+str(res$topLevel$result)
+
+str(res$topLevel$subclusterings)
+
+InSituType::flightpath_plot(
+  insitutype_result =
+    res$topLevel$subclusterings$structural$result)
```

Comment:
I would recommend to use  `res[["topLevel"]]$subclusterings[["structural"]]$result`, so it's clear which part of the results is the node names, which part is a fixed component.